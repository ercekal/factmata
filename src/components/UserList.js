import React from 'react';
import {isEmpty} from 'lodash';
import User from './User'

const UserList = ({data}) => {
  if (isEmpty(data)) {
    return (
      <div style={{margin: '16px'}}>There is no matching user</div>
    )
  }
  return (
    data.map(item => <User item={item} key={item.id}/>)
  );
};

export default UserList;