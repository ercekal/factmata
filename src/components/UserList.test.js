import React from 'react';
import UserList from './UserList';
import renderer from 'react-test-renderer';
import data from '../fixtures/userData.json'

it('renders data correctly', () => {
  const tree = renderer
    .create(<UserList data={data} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('shows error message if there is no data', () => {
  const data = []
  const tree = renderer
    .create(<UserList data={data} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});