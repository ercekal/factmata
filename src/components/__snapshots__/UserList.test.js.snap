// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="sc-bwzfXH ivwhAr"
    onClick={[Function]}
  >
    <div
      className="sc-htpNat eXoYXI"
    >
      <p
        className="sc-ifAKCX hbpRLv"
      >
        Leanne Graham
      </p>
    </div>
    <p
      className="sc-EHOje cwbZiP"
    >
      @
      Bret
    </p>
  </div>,
  <div
    className="sc-bwzfXH ivwhAr"
    onClick={[Function]}
  >
    <div
      className="sc-htpNat eXoYXI"
    >
      <p
        className="sc-ifAKCX hbpRLv"
      >
        Ervin Howell
      </p>
    </div>
    <p
      className="sc-EHOje cwbZiP"
    >
      @
      Antonette
    </p>
  </div>,
]
`;
